llvm/lib/SYCLPostLink/ModuleSplitter.cpp

```diff
@@ -1153,8 +1153,19 @@ std::string FunctionsCategorizer::computeCategoryFor(Function *F) const {
 std::unique_ptr<ModuleSplitterBase>
 getDeviceCodeSplitter(ModuleDesc &&MD, IRSplitMode Mode, bool IROutputOnly,
                       bool EmitOnlyKernelsAsEntryPoints) {
+  return getDeviceCodeSplitter(std::move(MD), Mode, IROutputOnly,
+                               EmitOnlyKernelsAsEntryPoints,
+                               AllowDeviceImageDependencies);
+}
+
+std::unique_ptr<ModuleSplitterBase>
+getDeviceCodeSplitter(ModuleDesc &&MD, IRSplitMode Mode, bool IROutputOnly,
+                      bool EmitOnlyKernelsAsEntryPoints,
+                      bool OverwriteAllowDeviceImageDependencies) {
   FunctionsCategorizer Categorizer;
 
+  AllowDeviceImageDependencies = OverwriteAllowDeviceImageDependencies;
```

Comment:
Please, describe a use-case that you want to cover.

This change looks controversial since the interface becomes sophisticated.

Reply:
`cl::opt<bool> AllowDeviceImageDependencies` can be removed and replaced by a member in `ModuleSplitterBase` class.

Reply:
https://github.com/intel/llvm/pull/18060 was merged. That should take care of this.